legacy/public/js/wavesurfer/libretime.js

```diff
@@ -0,0 +1,627 @@
+<defs><g id="K"><circle cx="32" cy="32" r="30" fill="${bg}"/>
+<line x1="32" y1="28" x2="32" y2="7" stroke-linecap="round" stroke-width="6" stroke="${fg}"/></g></defs>
+<use xlink:href="#K" transform="rotate(-135,32,32)"/>`;
+    for(let i=1;i<fr;++i)
+      r+=`<use xlink:href="#K" transform="translate(0,${64*i}) rotate(${-135+270*i/fr},32,32)"/>`;
+    return r+"</svg>";
+  }
+  let makeHSliderFrames=(fr,fg,bg,w,h)=>{
+    let r=
+`<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="${w}" height="${fr*h}" viewBox="0 0 ${w} ${fr*h}" preserveAspectRatio="none">
+<defs><g id="B"><rect x="0" y="0" width="${w}" height="${h}" rx="${h/2}" ry="${h/2}" fill="${bg}"/></g>
+<g id="K"><circle x="${w/2}" y="0" r="${h/2*0.9}" fill="${fg}"/></g></defs>`;
+    for(let i=0;i<fr;++i){
+      r+=`<use xlink:href="#B" transform="translate(0,${h*i})"/>`;
+      r+=`<use xlink:href="#K" transform="translate(${h/2+(w-h)*i/100},${h/2+h*i})"/>`;
+    }
+    return r+"</svg>";
+  }
+  let makeVSliderFrames=(fr,fg,bg,w,h)=>{
+    let r=
+`<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="${w}" height="${fr*h}" viewBox="0 0 ${w} ${fr*h}" preserveAspectRatio="none">
+<defs><rect id="B" x="0" y="0" width="${w}" height="${h}" rx="${w/2}" ry="${w/2}" fill="${bg}"/>
+<circle id="K" x="0" y="0" r="${w/2*0.9}" fill="${fg}"/></defs>`;
+    for(let i=0;i<fr;++i){
+      r+=`<use xlink:href="#B" transform="translate(0,${h*i})"/>`;
+      r+=`<use xlink:href="#K" transform="translate(${w/2} ${h*(i+1)-w/2-i*(h-w)/100})"/>`;
+    }
+    return r+"</svg>";
+  }
+  let initSwitches=(el)=>{
+    let w,h,d,fg,bg;
+    if(el.inputKnobs)
+      return;
+    el.inputKnobs={};
+    el.refresh=()=>{
+      let src=el.getAttribute("data-src");
+      d=+el.getAttribute("data-diameter");
+      let st=document.defaultView.getComputedStyle(el,null);
+      w=parseFloat(el.getAttribute("data-width")||d||st.width);
+      h=parseFloat(el.getAttribute("data-height")||d||st.height);
+      bg=el.getAttribute("data-bgcolor")||op.bgcolor;
+      fg=el.getAttribute("data-fgcolor")||op.fgcolor;
+      el.style.width=w+"px";
+      el.style.height=h+"px";
+      if(src)
+        el.style.backgroundImage="url("+src+")";
+      else {
+        let minwh=Math.min(w,h);
+        let svg=
+`<svg xmlns="http://www.w3.org/2000/svg" width="${w}" height="${h*2}" viewBox="0 0 ${w} ${h*2}" preserveAspectRatio="none">
+<g><rect fill="${bg}" x="1" y="1" width="${w-2}" height="${h-2}" rx="${minwh*0.25}" ry="${minwh*0.25}"/>
+<rect fill="${bg}" x="1" y="${h+1}" width="${w-2}" height="${h-2}" rx="${minwh*0.25}" ry="${minwh*0.25}"/>
+<circle fill="${fg}" cx="${w*0.5}" cy="${h*1.5}" r="${minwh*0.25}"/></g></svg>`;
+        el.style.backgroundImage="url(data:image/svg+xml;base64,"+btoa(svg)+")";
+      }
+    };
+    el.refresh();
+  };
+  let initKnobs=(el)=>{
```

Comment:
This is probably unused.

Reply:
initKnobs is used for the Zoom Slider. InitSwitches is indeed not used.